scripts/TestHarness/testers/RavenUtils.py

```diff
@@ -42,77 +42,130 @@ def in_python_3():
+                      ("matplotlib", 'matplotlib.__version__', "1.3.1", "2.2.3", None),
+                      ## NOTE there is a known bug in netCDF4 prior to 1.3.1 where
+                      # having a path length of exactly 88 characters can create a
+                      # seg fault.  However, h5py has no new libraries
+                      # after 2.7 and is not compatible with hdf5 greater than
+                      # 1.8.17, while netCDF4 requires hdf5 of at least 1.10.1.
+                      # For now, we avoid using netCDF
+                      # until we transition from
+                      # HDF5 databases and drop them like hot rocks.
+                      ("netCDF4", 'netCDF4.__version__', "1.2.3", "1.4.0", None), # 1.2.4
+                      # On Windows conda, there are no Python 2.7-compatible
+                      ## versions of TensorFlow, although
+                      ## these exist on Mac and Linux condas.  Darn.
+                      ("tensorflow", 'tensorflow.__version__', "1.12.0", "1.12.0", None),
+                      ("statsmodels", 'statsmodels.__version__', "0.8.0", "0.8.0", None)]
 
-def __lookup_preferred_version(name, optional=False):
+if in_python_3():
+  modules_to_try = modules_to_try_py3
+else:
+  modules_to_try = modules_to_try_py2
+
+def __lookup_preferred_version(name):
   """
     Look up the preferred version in the modules.
     @In, name, string, the name of the module
     @Out, result, string, returns the version as a string or "" if unknown
   """
-  index = modules_to_try if not optional else optional_test_libraries
-  for  mod_name, _, _, qa_ver, _ in index:
+  for  mod_name, _, _, qa_ver, _ in modules_to_try:
     if name == mod_name:
       return qa_ver
   return ""
+
 # libraries to install with Conda
-__condaList = [("h5py", __lookup_preferred_version("h5py")),
-               ("numpy", __lookup_preferred_version("numpy")),
+if not in_python_3():
+  __condaList = [("h5py", __lookup_preferred_version("h5py")),
+                 ("numpy", __lookup_preferred_version("numpy")),
+                 ("scipy", __lookup_preferred_version("scipy")),
+                 ("scikit-learn", __lookup_preferred_version("sklearn")),
+                 ("pandas", __lookup_preferred_version("pandas")),
+                 ("xarray", __lookup_preferred_version("xarray")),
+                 ("netcdf4", __lookup_preferred_version("netCDF4")),
+                 ("matplotlib", __lookup_preferred_version("matplotlib")),
+                 ("statsmodels", __lookup_preferred_version("statsmodels")),
+                 ("python", "2.7"),
+                 ("hdf5", "1.8.18"),
+                 ("swig", ""),
+                 ("pylint", ""),
+                 ("coverage", ""),
+                 ("lxml", ""),
+                 ("psutil", "")]
+  # libraries to install with conda-forge
+  __condaForgeList = [("pyside", "")]
+
+  __pipList = [("numpy", __lookup_preferred_version("numpy")),
```

Comment:
Did we decide the pip list should be maintained? If we're making that change here, we really need a machine that tests building with pip instead of conda.

Reply:
not valid